components/clp-package-utils/clp_package_utils/scripts/native/compress.py

```
@@ -153,41 +157,152 @@ def _generate_clp_io_config(
+    if len(keys) != len(set(keys)):
+        raise ValueError("Duplicate keys found in URLs")
+
+    # For a single key, use the full key as prefix with keys=None
+    # For multiple keys, find common prefix and provide explicit keys list
+    if len(keys) == 1:
+        key_prefix = keys[0]
+        keys = None
+    else:
+        key_prefix = os.path.commonprefix(keys)
+        if not key_prefix:
+            raise ValueError("No common prefix found among object keys")
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)